twofactor_privacyidea/lib/Controller/SettingsController.php

```diff
@@ -49,4 +51,44 @@ public function setValue($key, $value) {
 	public function getValue($key) {
 		return $this->config->getAppValue("twofactor_privacyidea", $key);
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Send a authentication request to privacyIDEA to check the connection.
+	 *
+	 * @param string $user user to send to privacyIDEA
+	 * @param string $password password to send to privacyIDEA
+	 */
+	public function testAuthentication($user, $pass) {
+		// instantiate our very own twofactor provider
+		$provider = OC::$server->query("OCA\TwoFactor_privacyIDEA\Provider\TwoFactorPrivacyIDEAProvider");
+		$status = "error";
+		try {
+			$result = $provider->authenticate($user, $pass);
+			if($result) {
+				$message = "Successfully authenticated!";
+				$status = "success";
+			} else {
+				$message = "Failed to authenticate.";
```

Comment:
Actually at this point I think we are testing, if the connection to privacyIDEA works OK. Users might not have tokens yet, so the admin could also test with an arbitrary password. Of course authentication would fail, but as a connection test, the test would be successful.

So I think we should change the wording here.

Something like:

"Communitcation to the privacyIDEA server succeeded. The user was successfully authenticated." in case of success.

"Communitcation to the privacyIDEA server succeeded. However, the user failed to authenticate."
For a connection test, this might still a valid result. (status=true, value=false).

I think in other cases, when an error occurs either on HTTP level or in privacyIDEA (``status=false``) we probably raise an exception?

So I suggest:
1. Change the wording
2. We might even set ``$status="success", even if the authentication fails?

Reply:
I think using an errorCode in the exception sounds good. Even for other aspects in the future. 
We might need to restructure codes later.  But for now it seems ok.